src/main/java/seedu/address/model/member/Gender.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.member;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Member's gender in CCACommander.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS = "Gender should only be one of the following: \n"
+            + "Male, Female, or Others";
+
+    public final String value;
+
+    /**
+     * Constructs an {@code Gender}.
+     *
+     * @param gender A valid email address.
+     */
+    public Gender(String gender) {
+        requireNonNull(gender);
+        checkArgument(isValidGender(gender), MESSAGE_CONSTRAINTS);
+        value = gender;
+    }
+
+    /**
+     * Returns if a given string is a valid email.
+     */
+    public static boolean isValidGender(String test) {
+        return test.equals("Male")
+                || test.equals("Female")
+                || test.equals("Others");
```

Comment:
should we do test.trim().equals() just for safety reasons to ensure no white space affect the result. Or is this intentional to ensure no whitespace is entered by user

Reply:
Ah that's a good point, will implement that

Reply:
Actually on second thought, should this be kept? Maybe we should have an exact match to ensure there is no whitespace entered as now we will accept `Male      ` and store it as `Male      ` instead of just `Male`

Reply:
I have updated the code to trim the gender string before storing it into the value param. Is this alright since the other classes don't do this?

Reply:
I think this is fine, we can think about doing this for the other classes as well.